girder/girder_large_image/web_client/vue/components/CompositeLayers.vue

```diff
@@ -47,15 +47,20 @@ export default {
                     autoRange: undefined
                 }
             })
-            Object.entries(CHANNEL_COLORS).forEach(([channelName, color]) => {
-                if(this.layers.includes(channelName)){
-                    this.compositeLayerInfo[channelName].palette = color
-                    usedColors.push(color)
-                }
-            })
+            // Assign colors
             this.layers.forEach((layerName) => {
                 if (!this.compositeLayerInfo[layerName].palette) {
                     let chosenColor;
+
+                    // Search for case-insensitive regex match among known channel-colors
+                    Object.entries(CHANNEL_COLORS).forEach(([channelName, color]) => {
+                        if (layerName.toUpperCase().match(
+                            new RegExp("^" + channelName + ".*$")
+                        )) {
+                            chosenColor = color
+                        }
+                    })
+
```

Comment:
Thanks for catching that. I just added an `if` block to protect it: [598c059](https://github.com/girder/large_image/pull/1222/commits/598c05955c29608887469dd6a7063e2d1411e078)

Reply:
Good, but looking at this again, I think we've changed some essential logic.  Before we were assigning colors to all the channels where we had specified a default color.  We need to still do this first, but avoid using a color more than once, because otherwise a file with multiple DAPI channels will make them all the same color.

Reply:
I wonder if we should just make the keys of the CHANNEL_COLORS a regex string. If we define various CD values, CD4 would also match CD41.

Reply:
Oh, I see. How about this: [4288a06](https://github.com/girder/large_image/pull/1222/commits/4288a0613960a17772ec3a63fceb81c20befb70c)

Reply:
Yes.  I'd adjust the ones that end in numbers to something like `'^CY3(|[^\d].*)$'`. (that is, either end immediately or have one non-digit character followed by anything else).

Reply:
Ok, changed in [cd08c1c](https://github.com/girder/large_image/pull/1222/commits/cd08c1c60d0d2dab6baaf97bc98811448191e147)